worlds/witness/hints.py

```diff
@@ -382,44 +447,55 @@ def hint_from_location(world: "WitnessWorld", location: str) -> Optional[Witness
     return WitnessLocationHint(location_obj, True)
 
 
-def get_items_and_locations_in_random_order(world: "WitnessWorld", own_itempool: List[Item]):
-    prog_items_in_this_world = sorted(
+def get_item_and_location_names_in_random_order(world: "WitnessWorld", own_itempool: List[Item]):
+    prog_item_names_in_this_world = sorted(
         item.name for item in own_itempool
         if item.advancement and item.code and item.location
     )
+    world.random.shuffle(prog_item_names_in_this_world)
```

Comment:
If you're shuffling, you probably don't need to sort? Same for `locations_in_this_world` I think. They should be ordered already, so it shouldn't affect determinism.